File: saizad-mvvm/src/main/java/com/saizad/mvvm/components/form/ui/IntegerInputFieldView.java
package com.saizad.mvvm.components.form.ui;

import android.content.Context;
import android.util.AttributeSet;

import com.sa.easyandroidform.ObjectUtils;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class IntegerInputFieldView extends InputFieldView<Integer> {

    public IntegerInputFieldView(@NotNull Context context) {
        super(context);
    }

    public IntegerInputFieldView(@NotNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public IntegerInputFieldView(@NotNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }


    @Override
    public Integer resolveFrom(@NotNull CharSequence charSequence) {
        try {
            return Integer.valueOf(charSequence.toString());
        }catch (NumberFormatException e){
            return null;
        }
    }

    @Override
    public CharSequence resolve(@Nullable Integer value) {
        if(ObjectUtils.isNull(value)){
            return null;
        }
        return value.toString();
    }

    @Override
    public boolean isSame(Integer value, Integer prevValue) {
        return value.equals(prevValue);
    }
}

File: saizad-mvvm/src/main/java/com/saizad/mvvm/components/form/ui/IntegerInputFieldView.kt
package com.saizad.mvvm.components.form.ui

import android.content.Context
import android.util.AttributeSet
import com.sa.easyandroidform.ObjectUtils

open class IntegerInputFieldView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0
) : InputFieldView<Int>(context, attrs, defStyleAttr) {

    override fun resolveFrom(charSequence: CharSequence): Int? {
        return try {
            Integer.valueOf(charSequence.toString())
        } catch (e: NumberFormatException) {
            null
        }
    }

    override fun resolve(value: Int?): CharSequence? {
        return if (ObjectUtils.isNull(value)) {
            null
        } else value.toString()
    }

    override fun isSame(value: Int, prevValue: Int): Boolean {
        return value == prevValue
    }
}